chezmoi panics when destination directory is /

When destination directory is `/` chezmoi fails with `panic: /etc/somefile.conf: not in /"`. This used to work in pre-2.x days. If I have a `chezmoi.toml` as follows:

```bash
~ cat /etc/chezmoi.toml 
sourcedir = "/home/user/configrepo"
destdir   = "/"
```
And inside of `configrepo` I have:

```bash
~ ls -lR /home/user/configrepo 
/home/user/configrepo:
total 12
drwxr-xr-x 5 user user 4096 Oct  6  2020 etc

/home/user/configrepo/etc:
total 24
-rw-r--r-- 1 user user  259 Oct  6  2020 somefile.conf
```
Now, whether I use a `chezmoi.toml` as configured above or whether I specify the source and destination directories from the command line, I get this error when running:

```bash
~ sudo chezmoi -c /etc/chezmoi.toml --verbose
panic: /etc/somefile.conf: not in /                       
 
goroutine 1 [running]:
github.com/twpayne/chezmoi/v2/internal/chezmoi.AbsPath.MustTrimDirPrefix(...)
	github.com/twpayne/chezmoi/v2/internal/chezmoi/path.go:49
github.com/twpayne/chezmoi/v2/internal/chezmoi.(*GitDiffSystem).trimPrefix(...)
	github.com/twpayne/chezmoi/v2/internal/chezmoi/gitdiffsystem.go:232
github.com/twpayne/chezmoi/v2/internal/chezmoi.(*GitDiffSystem).encodeDiff(0xc0002bc720, 0xc00012a730, 0xe, 0xc0000e6800, 0x103, 0x200, 0x1a4, 0xc0000e6c00, 0xc0004a9d00)
	github.com/twpayne/chezmoi/v2/internal/chezmoi/gitdiffsystem.go:223 +0x659
github.com/twpayne/chezmoi/v2/internal/chezmoi.(*GitDiffSystem).WriteFile(0xc0002bc720, 0xc00012a730, 0xe, 0xc0000e6800, 0x103, 0x200, 0xc0000001a4, 0x0, 0x0)
	github.com/twpayne/chezmoi/v2/internal/chezmoi/gitdiffsystem.go:183 +0x73
github.com/twpayne/chezmoi/v2/internal/chezmoi.(*TargetStateFile).Apply(0xc00064eeb0, 0x559306a45660, 0xc0002bc720, 0x559306a417b8, 0xc0008c21a0, 0x559306a388a8, 0xc0003dfc80, 0x0, 0x559306a388a8, 0xc0003dfc80)
	github.com/twpayne/chezmoi/v2/internal/chezmoi/targetstateentry.go:125 +0x2b9
github.com/twpayne/chezmoi/v2/internal/chezmoi.(*SourceState).Apply(0xc00001a9a0, 0x559306a45660, 0xc0002bc720, 0x559306a45660, 0xc0002bc630, 0x559306a417b8, 0xc0008c21a0, 0x7fffe6f8b7d4, 0x1, 0xc00012a480, ...)
	github.com/twpayne/chezmoi/v2/internal/chezmoi/sourcestate.go:465 +0x402
github.com/twpayne/chezmoi/v2/internal/cmd.(*Config).applyArgs(0xc000240000, 0x559306a45660, 0xc0002bc720, 0x7fffe6f8b7d4, 0x1, 0xc0000f21e0, 0x0, 0x5, 0xc0008a7b58, 0x0, ...)
	github.com/twpayne/chezmoi/v2/internal/cmd/config.go:480 +0x337
github.com/twpayne/chezmoi/v2/internal/cmd.(*Config).runDiffCmd(0xc000240000, 0xc0002b8780, 0xc0000f21e0, 0x0, 0x5, 0x0, 0x0)
	github.com/twpayne/chezmoi/v2/internal/cmd/diffcmd.go:47 +0x1ff
github.com/spf13/cobra.(*Command).execute(0xc0002b8780, 0xc0008c9a00, 0x5, 0x8, 0xc0002b8780, 0xc0008c9a00)
	github.com/spf13/cobra@v1.1.3/command.go:852 +0x472
github.com/spf13/cobra.(*Command).ExecuteC(0xc000272280, 0xc000272280, 0x0, 0x0)
	github.com/spf13/cobra@v1.1.3/command.go:960 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.1.3/command.go:897
github.com/twpayne/chezmoi/v2/internal/cmd.(*Config).execute(0xc000240000, 0xc000032080, 0x6, 0x6, 0x0, 0x0)
	github.com/twpayne/chezmoi/v2/internal/cmd/config.go:873 +0x7f
github.com/twpayne/chezmoi/v2/internal/cmd.runMain(0x5593068453d0, 0x5, 0x0, 0x0, 0x5593068464f0, 0x14, 0x0, 0x0, 0xc000032080, 0x6, ...)
	github.com/twpayne/chezmoi/v2/internal/cmd/cmd.go:260 +0xd6
github.com/twpayne/chezmoi/v2/internal/cmd.Main(0x5593068453d0, 0x5, 0x0, 0x0, 0x5593068464f0, 0x14, 0x0, 0x0, 0xc000032080, 0x6, ...)
	github.com/twpayne/chezmoi/v2/internal/cmd/cmd.go:87 +0x90
main.main()
	github.com/twpayne/chezmoi/v2/main.go:23 +0x133
```
## chezmoi doctor

```bash
~ sudo chezmoi -c /etc/chezmoi.toml doctor
RESULT    CHECK                MESSAGE
warning   version              v2.1.0, built at 2021-07-04T08:45:50Z
ok        os-arch              linux/amd64 (Arch Linux/)
ok        config-file          /etc/chezmoi.toml is a file
ok        source-dir           /home/user/configrepo is a directory
ok        suspicious-entries   no suspicious entries found in /home/user/configrepo
ok        dest-dir             / is a directory
ok        shell                found /bin/bash
ok        editor               found /usr/bin/vi
ok        git-cli              found /usr/bin/git, version 2.32.0
ok        merge-cli            found /usr/bin/vimdiff
ok        age-cli              age not found in $PATH
ok        gnupg-cli            found /usr/bin/gpg, version 2.2.28
ok        1password-cli        found /usr/bin/op, version 1.10.3
info      bitwarden-cli        bw not found in $PATH
info      gopass-cli           gopass not found in $PATH
info      keepassxc-cli        keepassxc-cli not found in $PATH
info      keepassxc-db         not set
info      lastpass-cli         lpass not found in $PATH
info      pass-cli             pass not found in $PATH
info      vault-cli            vault not found in $PATH
info      secret-cli           not set
```

--- a/internal/chezmoi/path.go
+++ b/internal/chezmoi/path.go
@@ -77,13 +77,17 @@ func (p AbsPath) String() string {
 
 // TrimDirPrefix trims prefix from p.
 func (p AbsPath) TrimDirPrefix(dirPrefixAbsPath AbsPath) (RelPath, error) {
-	if !strings.HasPrefix(string(p), string(dirPrefixAbsPath+"/")) {
+	dirAbsPath := dirPrefixAbsPath
+	if dirAbsPath != "/" {
+		dirAbsPath += "/"
+	}
+	if !strings.HasPrefix(string(p), string(dirAbsPath)) {
 		return "", &errNotInAbsDir{
 			pathAbsPath: p,
 			dirAbsPath:  dirPrefixAbsPath,
 		}
 	}
-	return RelPath(p[len(dirPrefixAbsPath)+1:]), nil
+	return RelPath(p[len(dirAbsPath):]), nil
 }
 
 // Type implements github.com/spf13/pflag.Value.Type.

